evap/results/tools.py

```diff
@@ -160,11 +160,17 @@ def avg_distribution(distributions, weights=itertools.repeat(1)):
 
 
 def average_grade_questions_distribution(results):
-    return avg_distribution([normalized_distribution(result.counts) for result in results if result.question.is_grade_question])
+    return avg_distribution(
+        [normalized_distribution(result.counts) for result in results if result.question.is_grade_question],
+        [sum(result.counts) for result in results if result.question.is_grade_question]
+    )
 
 
 def average_non_grade_rating_questions_distribution(results):
-    return avg_distribution([normalized_distribution(result.counts) for result in results if result.question.is_non_grade_rating_question])
+    return avg_distribution(
+        [normalized_distribution(result.counts) for result in results if result.question.is_non_grade_rating_question],
+        [sum(result.counts) for result in results if result.question.is_non_grade_rating_question]
+    )
```

Comment:
it looks like all our calls to `avg_distribution` are now using weights. you can remove the default weights parameter of that function. could you please also consider whether it would make things easier if it took a list of `(distribution, weight)` tuples instead of two separate lists? i'm not sure myself, but it would at least remove the duplicate loops. in the call sites.